pkg/testing/kubernetes/loader.go

```diff
@@ -0,0 +1,69 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package kubernetes
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+
+	"github.com/elastic/cloud-on-k8s/v2/pkg/apis/agent/v1alpha1"
+)
+
+// LoadFromYAML converts the given YAML reader to a list of k8s objects
+func LoadFromYAML(reader *bufio.Reader) ([]k8s.Object, error) {
+	// if we need to encode/decode more k8s object types in our tests, add them here
+	k8sScheme := runtime.NewScheme()
+	k8sScheme.AddKnownTypes(rbacv1.SchemeGroupVersion, &rbacv1.ClusterRoleBinding{}, &rbacv1.ClusterRoleBindingList{})
+	k8sScheme.AddKnownTypes(rbacv1.SchemeGroupVersion, &rbacv1.ClusterRole{}, &rbacv1.ClusterRoleList{})
+	k8sScheme.AddKnownTypes(rbacv1.SchemeGroupVersion, &rbacv1.RoleBinding{}, &rbacv1.RoleBindingList{})
+	k8sScheme.AddKnownTypes(rbacv1.SchemeGroupVersion, &rbacv1.Role{}, &rbacv1.RoleList{})
+	k8sScheme.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.ServiceAccount{}, &corev1.ServiceAccountList{})
+	k8sScheme.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.Pod{}, &corev1.PodList{})
+	k8sScheme.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.Service{}, &corev1.ServiceList{})
+	k8sScheme.AddKnownTypes(appsv1.SchemeGroupVersion, &appsv1.DaemonSet{})
+	k8sScheme.AddKnownTypes(appsv1.SchemeGroupVersion, &appsv1.StatefulSet{})
+	k8sScheme.AddKnownTypes(appsv1.SchemeGroupVersion, &appsv1.Deployment{})
+	k8sScheme.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.Secret{}, &corev1.ConfigMap{})
+	k8sScheme.AddKnownTypes(schedulingv1.SchemeGroupVersion, &schedulingv1.PriorityClass{})
```

Comment:
To register all the builtin resources, you can use [k8s.io/client-go/kubernetes/scheme.AddToScheme](https://pkg.go.dev/k8s.io/client-go/kubernetes/scheme#pkg-variables).

Reply:
thanks for the proposal, didn't know that 🙂 pushed it here 0b0b16be37b460123ad95c0b7789cfcd8f2c6382. Feel free to unresolve if you spot anything else